pkg/controller/acmeorders/sync.go

```diff
@@ -109,7 +109,7 @@ func (c *controller) Sync(ctx context.Context, o *cmacme.Order) (err error) {
 	// TODO: is this state possible? Either remove this case or add a comment as to what path could lead to it
 	case o.Status.State == cmacme.Valid && o.Status.Certificate == nil:
 		log.V(logf.DebugLevel).Info("Order is in a Valid state but the Certificate data is empty, fetching existing Certificate")
-		return c.fetchCertificateData(ctx, cl, o)
+		return c.syncCertificateData(ctx, cl, o, genericIssuer)
```

Comment:
Renamed these functions as they not only fetch the cert, but also store it on Order's status